edx_exams/apps/core/models.py

```diff
@@ -181,14 +188,28 @@ def get_current_exam_attempt(cls, user_id, exam_id):
     @classmethod
     def get_attempt_by_id(cls, attempt_id):
         """
-        Return ExamAttempt for a given id
+        Return ExamAttempt for a given attempt_id
         """
         try:
             attempt = cls.objects.get(id=attempt_id)
         except cls.DoesNotExist:
             attempt = None
         return attempt
 
+    @classmethod
+    def get_latest_attempt_for_user(cls, user_id):
```

Comment:
Ok I dug around how the UI uses this a bit and have a suggestion. The frontend is always going to want to know when if an exam changes state and transitioning to submitted or error is a common occurrence. Simply not returning anything in those cases probably won't work as we have it right now. Based on how our UI works there's no danger in returning additional states here since it's always checking the state this endpoint returns. My suggestion, drop the status__in filter entirely and just return by latest start_time.

tdlr; return the latest active attempt regardless of the status of that attempt